### PR TITLE
fix: conversion of max memory allocated and reserved to GB

### LINKED ISSUE #120171: Conversion of Max memory allocated and reserved numbers in GB are wrong on export_memory_timeline() in profiler
### 🐛 Describe the bug

When exporting a memory timeline using `export_memory_timeline()` with a HTML file path something like `export_memory_timeline("./exp_trace.html")`. The output shows incorrect max memory allocated and max memory reserved in GB. 

For example the `torch.cuda.max_memory_allocated()` returns `81352720384 bytes` that should be around `75.77 GB` however, the timeline reports `81.35 GB` this is not be possible as I am using a 80GB memory GPU. 

Moving to GiB would be more clear.

This has to be fixed - https://github.com/pytorch/pytorch/blob/2864a7e161cc107f7e4c00cccdf860a6089c73c3/torch/profiler/_memory_profiler.py#L1180-L1181

### Versions

[pip3] numpy==1.24.3
[pip3] torch==2.1.2
[pip3] torchvision==0.15.1
[pip3] triton==2.1.0

cc @robieta @chaekit @aaronenyeshi @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov @davidberard98

### PR DESCRIPTION
**Pull request by kmehant**
Fixes #120171

### CODE REVIEW

File: torch/profiler/_memory_profiler.py [hunk 1]
```diff
@@ -1176,8 +1176,8 @@ def export_memory_timeline_html(
         title = "\n\n".join(
             ([title] if title else [])
             + [
-                f"Max memory allocated: {max_memory_allocated/(10**9):.2f} GB \n"
-                f"Max memory reserved: {max_memory_reserved/(10**9):.2f} GB"
+                f"Max memory allocated: {max_memory_allocated/(1024**3):.2f} GB \n"
```
Comment (eqy) [inline torch/profiler/_memory_profiler.py:1179]: Nit: we may want to switch to "GiB" for clarity here: https://en.wikipedia.org/wiki/Byte#Multiple-byte_units

Comment (kmehant) [reply torch/profiler/_memory_profiler.py:1179]: Thanks @eqy. I will update the PR. On a side note, I see GB and GiB are being used interchangeably though they are technically different. For instance, NVIDIA uses GB (A100 80GB (https://www.nvidia.com/en-in/data-center/a100/)) though they meant 80 GiB which is around 86GB. I will update the PR with GiB, I am sure that will be more clear. Thank you.

**Review by soulitzer — APPROVED** [review]
Thanks!

**Comment by soulitzer** [comment]
@kmehant looks like you need to sign CLA

**Review by aaronenyeshi — APPROVED** [review]
LGTM

**Comment by kmehant** [comment]
https://github.com/pytorch/pytorch/pull/120172#issuecomment-1953352488

This being my first contribution, I am working on the CLA part, will update here, as I get that done. Thank you.

**Comment by kmehant** [comment]
@aaronenyeshi @soulitzer I have completed the CLA. Thank you.

**Pull request closed without merging**